src/routes/resources/[resourceContentId]/+page.svelte

```diff
@@ -1019,128 +1023,128 @@
-                    <MachineTranslationRating
-                        {machineTranslationStore}
-                        showingInPrompt={true}
-                        improvementHorizontalPositionPx={0}
-                    />
-                </div>
-            </div>
-        {/if}
-        <UserSelector
-            users={usersThatCanBeAssigned()}
-            defaultLabel="Select User"
-            bind:selectedUserId={assignToUserId}
-            hideUser={resourceContent?.assignedUser}
-        />
-    </Modal>
-
-    <Modal
-        header={hasUnresolvedThreads && resourceContent?.status !== ResourceContentStatusEnum.New
-            ? 'Confirm Publish'
-            : 'Choose Publish Option'}
-        bind:open={isPublishModalOpen}
-        primaryButtonText="Publish"
-        primaryButtonOnClick={publish}
-        primaryButtonDisabled={$isPageTransacting}
-    >
-        {#if hasUnresolvedThreads && resourceContent?.status !== ResourceContentStatusEnum.New}
-            <p class="py-4 text-lg text-warning">This resource has unresolved comments.</p>
-        {/if}
-        {#if resourceContent?.status === ResourceContentStatusEnum.New}
-            <div class="form-control">
-                <label class="label cursor-pointer justify-start space-x-2">
-                    <input type="checkbox" bind:checked={createDraft} class="checkbox" />
-                    <span class="label-text">Aquiferization Needed</span>
-                </label>
-            </div>
-            <!-- svelte-ignore a11y-label-has-associated-control -->
-            <label class="form-control">
-                <div class="label">
-                    <span class="label-text">Aquiferization Assignment (optional)</span>
-                </div>
-                <UserSelector
-                    users={usersThatCanBeAssigned()}
-                    defaultLabel="Unassigned"
-                    disabled={!createDraft}
-                    bind:selectedUserId={assignToUserId}
-                />
-            </label>
-        {/if}
-    </Modal>
-
-    <Modal
-        header="Create Translation"
-        bind:open={isAddTranslationModalOpen}
-        primaryButtonText="Create"
-        primaryButtonOnClick={createTranslation}
-        primaryButtonDisabled={newTranslationLanguageId === null || $isPageTransacting}
-    >
-        <TranslationSelector
-            allLanguages={data.languages}
-            existingTranslations={resourceContent?.contentTranslations ?? []}
-            bind:selectedLanguageId={newTranslationLanguageId}
-        />
-        <div slot="additional-buttons">
-            {#if englishContentTranslation?.hasDraft}
-                <div>
-                    <label class="label cursor-pointer">
-                        <input
-                            type="checkbox"
-                            class="checkbox-primary checkbox me-2"
-                            bind:checked={createTranslationFromDraft}
+    {#key resourceContentId}
```

Comment:
The modals were moved into the template area that renders after the promise is completed. This allows for the additional filtering logic that needs a resource content status.